bin/detect

```diff
@@ -23,18 +23,22 @@
 BP=$(dirname "$(dirname "$0")")
 
 # Generally stacks do not have ruby or python installed, with the exception of cflinuxfs3, so we install them here
-# We deliberately skip re-installing python and ruby on the cflinuxfs3 stack to avoid compatibility issues
-# (e.g. the bootstrapped python is incompatible with cflinuxfs3 due to being compiled on a newer version of GBLIC)
+#
+# We skip re-installing ruby on the cflinuxfs3 stack to avoid compatibility issues
 if [ "$CF_STACK" != "cflinuxfs3" ]; then
-  PYTHON_DIR="/tmp/php-buildpack/python"
-  mkdir -p "${PYTHON_DIR}"
-  source "$BP/bin/install-python" "$PYTHON_DIR" "$BP" &> /dev/null
-
   RUBY_DIR="/tmp/php-buildpack/ruby"
   mkdir -p "${RUBY_DIR}"
   source "$BP/bin/install-ruby" "$RUBY_DIR" "$BP" &> /dev/null
 fi
 
+# To avoid having to support both python 2 & 3 and to avoid using the ancient
+# python included in cflinuxfs3, always install python, unless running unit tests
+if [ -z "${USE_SYSTEM_PYTHON}" ]; then
+  PYTHON_DIR="/tmp/php-buildpack/python"
+  mkdir -p "${PYTHON_DIR}"
+  source "$BP/bin/install-python" "$PYTHON_DIR" "$BP" &> /dev/null
+fi
+
 export PYTHONPATH=$BP/lib
 VERSION="$(cat "$BP"/VERSION)"
-python2 "$BP/"scripts/detect.py "$1" "$VERSION"
+python "$BP/"scripts/detect.py "$1" "$VERSION"
```

Comment:
If we `if [ -z "${USE_SYSTEM_PYTHON}" ]` doesn't execute, does this line work running just `python`?

Reply:
It will use whatever `python` is on your path. There are a few branching paths:

For unit tests, it should use whatever you're running the tests with. 
- For local python tests (`./run_test.sh` or `nosetests ...`), the recommended approach from the README is  installed via `pyenv`, with the version specified by `.python-version`
- For running the tests in Docker (`./scripts/unit.sh` or `go test ...` with a docker daemon running), I made it so that [it installs python using `bin/install-python`](https://github.com/cloudfoundry/php-buildpack/blob/4e70f5dfacd8f3b0d75c8c1f9a82d59068aabc13/src/php/unit/python_unit_specs_test.go#L31)

If the buildpack is actually installed on a CF (for integration tests or in production usage), then this environment variable should not be set.